doc/sphinx/practical-tools/utilities.rst

```diff
@@ -271,76 +272,83 @@ In :cmd:`Require` commands referring to the current package (if `_CoqProject`
-  is the logical name followed by a suffix of the physical path (as for `Coq`
-  above).  Use these options directly if you must.  Command line options are described
-  :ref:`here <command-line-options>`.
-
-- Using the command line options given in the `_CoqProject` file.
-  When you compile a file with `coqc` or step through a file with CoqIDE or
-  the other IDEs, they automatically look for a `_CoqProject` file in a
-  parent directory of the script file, and, if found, they apply the parameters from
-  `_CoqProject`.  Note that the `_CoqProject` file is ignored when Coq loads
-  compiled files from the package; every `.vo` file under the package directory is loaded.
-
-  Coq tools are limited to using only one `_CoqProject` at a time, so multi-package
-  projects need another mechanism.
-
-.. _COQPATH:
-
-- The `COQPATH` environment variable, which contains a list of directories.
-  Each of their subdirectories is a separately-compiled package.  The entire
-  logical path is derived from the directory name (as for `Bignums` above).
-  See :ref:`customization-by-environment-variables`.
-
-  .. todo improve what's written at the hyperlink
-
-- :cmd:`Add LoadPath` adds a new logical path/physical path pair.  We recommend
-  you avoid using :cmd:`Add LoadPath` because embedding physical paths in your scripts
-  makes it harder to install and use them on another computer.
-
-We recommend using `COQPATH`.  For simpler developments, it could have a single
-entry pointing to a directory that contains your various separate projects,
-each in its own subdirectory.  For more advanced
-development work, you may want to set `COQPATH` differently for different projects
-or different versions of a project.  Currently, if you want to change `COQPATH`
-while you're using CoqIDE, you must restart CoqIDE or start another CoqIDE process
-after making the change.  Perhaps in the future CoqIDE will show the current
-setting for and allow changing `COQPATH`.
-
-  .. note ::
-     Packages that have only the directory name ``.`` in ``-R <dirname> <dirpath>``
-     have logical names that map exactly to the directory names.  Including such
-     packages in a directory in `COQPATH` will work fine.
-
-     In contrast, packages that have a directory name other than `.` in `-R`
-     have subdirectories that don't become part of the :term:`logical path`.  If they're
-     installed in the Coq directories, they're fine.  If they're in `COQPATH`,
-     the `_CoqProject` file is essential to getting the correct logical paths within the packages.
-     Since Coq tools currently support only  a single `_CoqProject` file, such packages
-     work correctly only if the IDE is running a script that's part of that package.
-
-     Perhaps Coq will eventually support multiple `_CoqProject` files at once to eliminate
-     this limitation.
+Note that if you use a `_CoqProject` file, the `COQPATH` environment variable is not helpful.
```

Comment:
This is not strictly true as _CoqProject has multiple functions. One of them is passing command line arguments to coqc in certain subdirectories which COQPATH has nothing to do with. COQPATH simply prevents you from having to specify external projects that you depend on in your _CoqProject file which is already the case for anything installed in `lib/coq/user-contrib` as Coq is able to find and use those projects itself.

Reply:
> COQPATH simply prevents you from having to specify external projects that you depend on in your _CoqProject file

You can't just add the top source code directory into COQPATH.  For most packages, the logical names will be incorrect.  For example, `_CoqProject` for the equations package has:

```
-Q theories Equations
```

so you'd end up with an extraneous `theories` in the fully qualified logical names.  You would need to provide the additional parameter `-Q <source tree root dir>/theories` to make it load properly from other packages, and then you'd see two logical names for each directory (one with `theories` and one without) in `Print LoadPath`.  So not quite trivial to set up, and IMHO pretty much everyone is bad at configuration and dislikes doing it.

I think editing _CoqProject is the least bad alternative at the moment.  We'd probably be better off putting the list of uninstalled external packages in a separate (system-specific) file.  Maybe configure that in the IDE.  That file could also keep track of what files you have open for the package/project, the visible part of each buffer, etc.